libraries/AP_HAL_ChibiOS/GPIO.cpp

```diff
@@ -366,3 +383,22 @@ bool GPIO::wait_pin(uint8_t pin, INTERRUPT_TRIGGER_TYPE mode, uint32_t timeout_u
 
     return msg == MSG_OK;
 }
+
+/*
+  timer to setup setup interrupt quotas for a 100ms period from
+  monitor thread
+*/
+void GPIO::timer_tick()
+{
+    // allow 100k interrupts/second max for GPIO interrupt sources, which is
+    // 10k per 100ms call to timer_tick()
+    const uint16_t quota = 10000U;
+    for (uint8_t i=0; i<ARRAY_SIZE(_gpio_tab); i++) {
+        if (_gpio_tab[i].isr_quota == 1) {
+            // we ran out of ISR quota for this pin since the last
+            // check
+            INTERNAL_ERROR(AP_InternalError::error_t::gpio_isr);
```

Comment:
This is not an internal error, is it?

It's something that's caused by external things, not code bugs.

The description of what InternalError ought to be used for is in the header.

That probably just means adding a really big comment above this use of InternalError when it comes to it.  We may come back and find some other way of getting this information to the user at some stage.

Reply:
comment added